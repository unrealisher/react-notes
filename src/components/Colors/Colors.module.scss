.colors {
  display: flex;
  align-items: center;
  margin-bottom: 16px;
}

.title {
  font-size: 0;

  @media (min-width: 516px) {
    line-height: 42px;
    font-size: 36px;
    margin-right: 16px;
  }
}

.wrapper {
  width: 100%;
  overflow: hidden;
  height: 48px;

  @media (min-width: 516px) {
    height: 32px;
  }
}

.list {
  display: flex;
  padding-bottom: 16px;
  width: calc(100vw - 32px);
  overflow: auto;

  @media (min-width: 516px) {
    width: 320px;
  }
}

.item:not(:last-child) {
  margin-right: 16px;
}
