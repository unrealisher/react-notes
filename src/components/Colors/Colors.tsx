import React from "react";

import Color from "./../Color/Color";

import IColor from "./../../intefaces/IColor";

import styles from "./Colors.module.scss";

interface IProps {
  colors: IColor[];
}

const Colors = (props: IProps): JSX.Element => {
  const { colors } = props;
  return (
    <div className={styles.colors}>
      <h2 className={styles.title}>Заметки</h2>
      <div className={styles.wrapper}>
        <ul className={styles.list}>
          {colors.map(
            (item): JSX.Element => {
              const color: string = item.color;
              return (
                <li key={color} className={styles.item}>
                  <Color color={color} />
                </li>
              );
            }
          )}
        </ul>
      </div>
    </div>
  );
};

export default Colors;
