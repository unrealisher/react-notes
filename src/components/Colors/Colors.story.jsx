import React from "react";
import { Provider } from "react-redux";
import store from "./../../store/index";

import Colors from "./Colors";

import { storiesOf } from "@storybook/react";

import "./../../index.scss";

storiesOf("Colors", module).add("Стандарт", () => (
  <Provider store={store}>
    <Colors />
  </Provider>
));
