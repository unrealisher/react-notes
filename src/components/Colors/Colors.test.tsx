import React from "react";
import { shallow } from "enzyme";
import { Colors } from "./Colors";

describe("Colors", () => {
  const colors = [
    { id: 0, color: "#E84747" },
    { id: 1, color: "#F2994A" },
    { id: 2, color: "#F2C94C" },
    { id: 3, color: "#219653" },
    { id: 4, color: "#56CCF2" },
    { id: 5, color: "#2F80ED" },
    { id: 6, color: "#9B51E0" }
  ];
  it("correct render", () => {
    const component = shallow(<Colors colors={colors} />);
    expect(component).toMatchSnapshot();
  });
});
