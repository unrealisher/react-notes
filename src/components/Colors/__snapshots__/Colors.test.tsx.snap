// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Colors
    colors={
      Array [
        Object {
          "color": "#E84747",
          "id": 0,
        },
        Object {
          "color": "#F2994A",
          "id": 1,
        },
        Object {
          "color": "#F2C94C",
          "id": 2,
        },
        Object {
          "color": "#219653",
          "id": 3,
        },
        Object {
          "color": "#56CCF2",
          "id": 4,
        },
        Object {
          "color": "#2F80ED",
          "id": 5,
        },
        Object {
          "color": "#9B51E0",
          "id": 6,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
        >
          Заметки
        </h2>,
        <div
          className="wrapper"
        >
          <ul
            className="list"
          >
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#E84747",
                    "id": 0,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#F2994A",
                    "id": 1,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#F2C94C",
                    "id": 2,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#219653",
                    "id": 3,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#56CCF2",
                    "id": 4,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#2F80ED",
                    "id": 5,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#9B51E0",
                    "id": 6,
                  }
                }
              />
            </li>
          </ul>
        </div>,
      ],
      "className": "colors",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Заметки",
          "className": "title",
        },
        "ref": null,
        "rendered": "Заметки",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="list"
          >
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#E84747",
                    "id": 0,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#F2994A",
                    "id": 1,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#F2C94C",
                    "id": 2,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#219653",
                    "id": 3,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#56CCF2",
                    "id": 4,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#2F80ED",
                    "id": 5,
                  }
                }
              />
            </li>
            <li
              className="item"
            >
              <Connect(Color)
                color={
                  Object {
                    "color": "#9B51E0",
                    "id": 6,
                  }
                }
              />
            </li>
          </ul>,
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#E84747",
                      "id": 0,
                    }
                  }
                />
              </li>,
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#F2994A",
                      "id": 1,
                    }
                  }
                />
              </li>,
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#F2C94C",
                      "id": 2,
                    }
                  }
                />
              </li>,
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#219653",
                      "id": 3,
                    }
                  }
                />
              </li>,
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#56CCF2",
                      "id": 4,
                    }
                  }
                />
              </li>,
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#2F80ED",
                      "id": 5,
                    }
                  }
                />
              </li>,
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#9B51E0",
                      "id": 6,
                    }
                  }
                />
              </li>,
            ],
            "className": "list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "#E84747",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Color)
                  color={
                    Object {
                      "color": "#E84747",
                      "id": 0,
                    }
                  }
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": Object {
                    "color": "#E84747",
                    "id": 0,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "#F2994A",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Color)
                  color={
                    Object {
                      "color": "#F2994A",
                      "id": 1,
                    }
                  }
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": Object {
                    "color": "#F2994A",
                    "id": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "#F2C94C",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Color)
                  color={
                    Object {
                      "color": "#F2C94C",
                      "id": 2,
                    }
                  }
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": Object {
                    "color": "#F2C94C",
                    "id": 2,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "#219653",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Color)
                  color={
                    Object {
                      "color": "#219653",
                      "id": 3,
                    }
                  }
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": Object {
                    "color": "#219653",
                    "id": 3,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "#56CCF2",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Color)
                  color={
                    Object {
                      "color": "#56CCF2",
                      "id": 4,
                    }
                  }
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": Object {
                    "color": "#56CCF2",
                    "id": 4,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "#2F80ED",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Color)
                  color={
                    Object {
                      "color": "#2F80ED",
                      "id": 5,
                    }
                  }
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": Object {
                    "color": "#2F80ED",
                    "id": 5,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "#9B51E0",
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Color)
                  color={
                    Object {
                      "color": "#9B51E0",
                      "id": 6,
                    }
                  }
                />,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": Object {
                    "color": "#9B51E0",
                    "id": 6,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
          >
            Заметки
          </h2>,
          <div
            className="wrapper"
          >
            <ul
              className="list"
            >
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#E84747",
                      "id": 0,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#F2994A",
                      "id": 1,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#F2C94C",
                      "id": 2,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#219653",
                      "id": 3,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#56CCF2",
                      "id": 4,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#2F80ED",
                      "id": 5,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#9B51E0",
                      "id": 6,
                    }
                  }
                />
              </li>
            </ul>
          </div>,
        ],
        "className": "colors",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Заметки",
            "className": "title",
          },
          "ref": null,
          "rendered": "Заметки",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="list"
            >
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#E84747",
                      "id": 0,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#F2994A",
                      "id": 1,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#F2C94C",
                      "id": 2,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#219653",
                      "id": 3,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#56CCF2",
                      "id": 4,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#2F80ED",
                      "id": 5,
                    }
                  }
                />
              </li>
              <li
                className="item"
              >
                <Connect(Color)
                  color={
                    Object {
                      "color": "#9B51E0",
                      "id": 6,
                    }
                  }
                />
              </li>
            </ul>,
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="item"
                >
                  <Connect(Color)
                    color={
                      Object {
                        "color": "#E84747",
                        "id": 0,
                      }
                    }
                  />
                </li>,
                <li
                  className="item"
                >
                  <Connect(Color)
                    color={
                      Object {
                        "color": "#F2994A",
                        "id": 1,
                      }
                    }
                  />
                </li>,
                <li
                  className="item"
                >
                  <Connect(Color)
                    color={
                      Object {
                        "color": "#F2C94C",
                        "id": 2,
                      }
                    }
                  />
                </li>,
                <li
                  className="item"
                >
                  <Connect(Color)
                    color={
                      Object {
                        "color": "#219653",
                        "id": 3,
                      }
                    }
                  />
                </li>,
                <li
                  className="item"
                >
                  <Connect(Color)
                    color={
                      Object {
                        "color": "#56CCF2",
                        "id": 4,
                      }
                    }
                  />
                </li>,
                <li
                  className="item"
                >
                  <Connect(Color)
                    color={
                      Object {
                        "color": "#2F80ED",
                        "id": 5,
                      }
                    }
                  />
                </li>,
                <li
                  className="item"
                >
                  <Connect(Color)
                    color={
                      Object {
                        "color": "#9B51E0",
                        "id": 6,
                      }
                    }
                  />
                </li>,
              ],
              "className": "list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "#E84747",
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Color)
                    color={
                      Object {
                        "color": "#E84747",
                        "id": 0,
                      }
                    }
                  />,
                  "className": "item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": Object {
                      "color": "#E84747",
                      "id": 0,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "#F2994A",
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Color)
                    color={
                      Object {
                        "color": "#F2994A",
                        "id": 1,
                      }
                    }
                  />,
                  "className": "item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": Object {
                      "color": "#F2994A",
                      "id": 1,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "#F2C94C",
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Color)
                    color={
                      Object {
                        "color": "#F2C94C",
                        "id": 2,
                      }
                    }
                  />,
                  "className": "item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": Object {
                      "color": "#F2C94C",
                      "id": 2,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "#219653",
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Color)
                    color={
                      Object {
                        "color": "#219653",
                        "id": 3,
                      }
                    }
                  />,
                  "className": "item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": Object {
                      "color": "#219653",
                      "id": 3,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "#56CCF2",
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Color)
                    color={
                      Object {
                        "color": "#56CCF2",
                        "id": 4,
                      }
                    }
                  />,
                  "className": "item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": Object {
                      "color": "#56CCF2",
                      "id": 4,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "#2F80ED",
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Color)
                    color={
                      Object {
                        "color": "#2F80ED",
                        "id": 5,
                      }
                    }
                  />,
                  "className": "item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": Object {
                      "color": "#2F80ED",
                      "id": 5,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "#9B51E0",
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(Color)
                    color={
                      Object {
                        "color": "#9B51E0",
                        "id": 6,
                      }
                    }
                  />,
                  "className": "item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "color": Object {
                      "color": "#9B51E0",
                      "id": 6,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Colors not render when colors is empty array 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Colors
    colors={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
        >
          Заметки
        </h2>,
        <div
          className="wrapper"
        >
          <ul
            className="list"
          />
        </div>,
      ],
      "className": "colors",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Заметки",
          "className": "title",
        },
        "ref": null,
        "rendered": "Заметки",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="list"
          />,
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
          >
            Заметки
          </h2>,
          <div
            className="wrapper"
          >
            <ul
              className="list"
            />
          </div>,
        ],
        "className": "colors",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Заметки",
            "className": "title",
          },
          "ref": null,
          "rendered": "Заметки",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="list"
            />,
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "list",
            },
            "ref": null,
            "rendered": Array [],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
